dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/ClassDumper.java

```diff
@@ -395,4 +402,20 @@ URL findClassLocation(String className) throws ClassNotFoundException {
     }
     return javaClasses.build();
   }
+
+  /**
+   * Returns true if two class names (binary name JLS 13.1) have the same package.
+   */
+  static boolean classesInSamePackage(String classNameA, String classNameB) {
+    if (classNameA.contains(".") != classNameB.contains(".")) {
+      return false;
+    }
+    if (classNameA.lastIndexOf('.') < 0 && classNameB.lastIndexOf('.') < 0) {
+      return true;
+    }
+    // The conditions above ensure that lastIndexOf returns non-negative number
```

Comment:
The last index of '.' has to be the same (even if negative) for this to return true, right? Could you replace lines 410 – 415 with

```java
if (classNameA.lastIndexOf('.') != classNameB.lastIndexOf('.')) {
  return false;
}
```

Reply:
Updated the logic with introducing variables for className{A, B}.lastIndexOf('.')